server/src/routes/auth/login.ts

```diff
@@ -2,67 +2,78 @@ import { Request, Response } from 'express';
 import { prisma } from '../../config/db';
 import Joi from 'joi';
 import jwt from 'jsonwebtoken';
-import { jwtSecret } from '../../config/env';
+import { jwtSecret, mailSecret } from '../../config/env';
 import * as argon2id from 'argon2';
+import { sendMail } from '../../utils/mailer';
 import _ from 'lodash';
 
 export const login = {
-
     GET: (req: Request, res: Response) => {
         res.status(200).send('ok');
     },
 
     POST: [
-        (req: Request, res: Response) => {
+        async (req: Request, res: Response) => {
         let userInputScheme = Joi.object(({
             email: Joi.string().email().required(),
             name: Joi.string().normalize().required(),
             password: Joi.string().required()
         }));
         let userInput = userInputScheme.validate(req.body);
 
-        console.log(userInput);
-        if (userInput.error !== undefined) {
-            return res.status(400).json(userInputScheme.validate(req.body).error);
-        }
-        prisma.user.findUnique({
+        if (userInput.error !== undefined)
+            return res.status(400).json(userInput.error);
+
+        const user = await prisma.user.findFirst({
```

Comment:
Isn't the user supposed to be unique in db ?

```suggestion
        const user = await prisma.user.findUnique({
            where: {
                email: userInput.value.email,
                emailVerified: true
            }
        })
```

Reply:
For reference
https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#findunique

Reply:
Unfortunatly, `findUnique` only allows using parameters that has the tag  `@unique` in the schema definition in the where clause.
Since `emailVerified` is not unique (that wouldn't make any sense since it's a boolean), we can't use `findUnique`.
Howerver, since email is indeed an unique field, search by it would'nt cause too much trouble

Reply:
Ok, conversation resolved